Revised the `Style Guide` (#1794)
o3de/o3de.org: content/docs/contributing/to-docs/style-guide/quick-reference.md

---
linkTitle: Quick Reference
title: "Style Guide: Quick Reference"
description: A condensed reference of the Style Guide for contributors to the Open 3D Engine (O3DE) documentation.  
weight: 150
toc: true
---

Follow the O3DE Docs Style Guide in your documentation. This ensures that the documentation is consistent and helpful to users. 

| **Language** |  |
| --- | --- |

> | [Write for accessibility](guidance#write-for-accessibility) | Documentation is written in U.S. English. Use simple and precise words, wherever possible. Write short and complete sentences. Break up information into smaller bodies of text. |

Reviewer: ```suggestion
| [Write for accessibility](guidance#write-for-accessibility) | Documentation is written in U.S. English. Use simple and precise words wherever possible. Write short and complete sentences. Break up information into smaller bodies of text. |
```

> | [Write for accessibility](guidance.md#write-for-accessibility) | Documentation is written in U.S. English. Use simple and precise words, wherever possible. Write short and complete sentences. Break up information into smaller bodies of text. |

Author: Sure! Can you explain again why there's a preference to omit `.md`? If there is one, or if it's purely style.

> | [Write for accessibility](guidance.md#write-for-accessibility) | Documentation is written in U.S. English. Use simple and precise words, wherever possible. Write short and complete sentences. Break up information into smaller bodies of text. |

Reviewer 2: There was a time in the past when links would fail to resolve when `.md` was included. It's also less typing.

> | [Write for accessibility](guidance.md#write-for-accessibility) | Documentation is written in U.S. English. Use simple and precise words, wherever possible. Write short and complete sentences. Break up information into smaller bodies of text. |

Author: I omitted them in the next commit. However, my argument for pro-`.md` is that the links will work when I'm traversing through files in the GitHub repo. The use case for this is primarily for docs development, not so much customers.

> | [Voice and tone](guidance#voice-and-tone) | Use [active voice](https://writing.wisc.edu/handbook/style/ccs_activevoice/) and present-tense verbs wherever possible. Speak simply, respectfully, and professionally.  <br><br>Refer to the user as "you". Refer to the O3DE software, O3DE community, or the O3D Foundation as "we". |

Reviewer: ```suggestion
| [Voice and tone](guidance#voice-and-tone) | Use [active voice](https://writing.wisc.edu/handbook/style/ccs_activevoice/) and present-tense verbs wherever possible. Write simply, respectfully, and professionally.  <br><br>Refer to the user as "you". Refer to the O3DE software, O3DE community, or the O3D Foundation as "we". |
```

| [O3DE terminology](../terminology#o3de-specific-terms) | Use O3DE-specific terms as defined by the O3DE Documentation Terminology and Usage. Update the terminology if you introduce new terms. |
| [Standard domain and industry terminology](../terminology#standard-domain-and-industry-terminology) | Use terminology that is standard in the domain or industry. If there's ambiguity, then write context to make sure the reader understands the meaning. |
| [Terms to avoid](../terminology#terms-to-avoid-and-their-alternatives) | Aim for inclusivity in your choice of words. |
| [Acronyms, abbreviations, and Latin phrases](guidance#acronyms-abbreviations-and-latin-phrases) | Write acronyms or abbreviations in their expanded form first, followed by the acronym in parentheses. Don't abbreviate common words or Latin phrases; use the complete word or a similar one.  |
| [Idioms, slang, colloquialisms, or jargon](guidance#idioms-slang-colloquialisms-or-jargon) | Don't use them. These words and phrases may be understood by native U.S. English speakers, but are difficult to translate.  |
| **Formatting** |

> | [Hugo](../hugo) and [Markdown](format) | Documentation is written in Markdown (``) files and built by [Hugo](https://gohugo.io/), a static site generator. It's primarily formatted using [Markdown syntax](https://www.markdownguide.org/basic-syntax/), with support for in-line HTML. Additional content formatting includes images, videos, and callout boxes. Callout boxes are implemented through [shortcodes](shortcodes). |

Author: Fixed `.md`!
Also not seeing what you're seeing regarding the pipe `|`. 
![image](https://user-images.githubusercontent.com/75444793/180574390-981fb28b-6722-4555-85e4-37f5a5475a41.png)

> | [Hugo](../hugo) and [Markdown](format) | Documentation is written in Markdown (``) files and built by [Hugo](https://gohugo.io/), a static site generator. It's primarily formatted using [Markdown syntax](https://www.markdownguide.org/basic-syntax/), with support for in-line HTML. Additional content formatting includes images, videos, and callout boxes. Callout boxes are implemented through [shortcodes](shortcodes). |

Reviewer: Oh weird.  Maybe we see how it looks live...for science?

![formatting-wierdness](https://user-images.githubusercontent.com/31811688/180575264-c7865905-45a3-4f98-a32b-e3805ced15db.png)

| [Metadata](metadata) | Documentation must comply with metadata requirements, including *linktitle*, *title*, and *description*. Use *weight* to override the default alphabetical sort order. |

> | [Topic headings](format.md#topic-headings) | Section titles must use H2 (`##`) headings. Subsection titles follow as H3 (`###`) and H4 (`####`) headings. Topic headings must use sentence case. The H1(`#`) heading is reserved for the for the page title which is specified in the topic metadata.  |

Reviewer 3: ```suggestion
| [Topic headings](format.md#topic-headings) | Section titles must use H2 (`##`) headings. Subsection titles follow as H3 (`###`) and H4 (`####`) headings. Topic headings must use sentence case. The H1(`#`) heading is reserved for the page title which is specified in the topic metadata.  |
```

> | [Topic headings](format.md#topic-headings) | Section titles must use H2 (`##`) headings. Subsection titles follow as H3 (`###`) and H4 (`####`) headings. Topic headings must use sentence case. The H1(`#`) heading is reserved for the page title which is specified in the topic metadata.  |

Reviewer: ```suggestion
| [Topic headings](format.md#topic-headings) | Section titles must use H2 (`##`) headings. Subsection titles follow as H3 (`###`) and H4 (`####`) headings. Topic headings must use sentence case. The H1(`#`) heading is reserved for the page title which is specified in the topic's metadata.  |
```

| [User interface, inputs, and hotkeys](format#user-interface-inputs-and-hotkeys) | Bold all instances. |

> | [Files, directories, and paths](format#files-directories-and-paths) | Use `code-style` formatting. Provide context on what the path is relative to. When applicable, link to the file in the [o3de repository](https://github.com/o3de/o3de).  |

Reviewer: ```suggestion
| [Files, directories, and paths](format#files-directories-and-paths) | Use `code-style` formatting. Provide context on what a path is relative to. When applicable, link to the file in the [o3de repository](https://github.com/o3de/o3de).  |
```

> | [Applications, tools, Gems, and components](format/#applications-tools-gems-and-components) | Bold only the first on-page appearance. Subsequent appearances on the same page remain unformatted.  |

Reviewer: ```suggestion
| [Applications, tools, Gems, and components](format/#applications-tools-gems-and-components) | Bold only the first on-page occurrence. Subsequent occurrences on the same page remain unformatted.  |
```

> | [Code, commands, and API objects](format#code-commands-and-apis) | Use `code-style` formatting for commands and API objects. Use code blocks for multiple lines of code. For API documentation, you may need to use advanced formatting -- see [[Formatting for API Documentation]]. |

Author: Ah, that's referring to something I have in draft! Removed he whole sentence.

> | [Terminology](format.md/#terminology) | Be consistent in the use of terms. Italicize new terms, and follow with definition. Terminology that's unique to O3DE must be included in the Terminology page.  |

Reviewer 3: ```suggestion
| [Terminology](format.md/#terminology) | Be consistent in the use of terms. Italicize new terms, and follow with a definition. Terminology that's unique to O3DE must be included in the [Terminology page](../terminology#o3de-specific-terms).  |
```

> | [Terminology](format/#terminology) | Be consistent in the use of terms. Italicize new terms, and follow with a definition. Terminology that's unique to O3DE must be included in the [Terminology page](../terminology#o3de-specific-terms). |

Reviewer: ```suggestion
| [Terminology](format/#terminology) | Be consistent in the use of terms. Italicize new terms, and follow with a definition. Terms that are unique to O3DE must be included in the [Terminology page](../terminology#o3de-specific-terms). |
```
For consistency.

| [Trademark terminology](format.md/#trademark) | Properly format trademark titles and terminology according to its use from the source. Provide a link to the source's relevant material. |
| **Additional content** |
| [Images](media.md#adding-images-with-markdown-syntax) | Static images must be `.png` format. Diagrams must be `.svg` format. |

> | [Animated images](media.md#animated-images) | Use short videos (of a few seconds) to demonstrate a feature's functionality. Prefer to use `.mp4` files that are less than 512 KB. `.ogg` and `.webm` formats, and up to 1 MB of size are also accepted. `.gif` format is not permitted. |

Reviewer 3: ```suggestion
| [Animated images](media.md#animated-images) | Use short videos (of a few seconds) to demonstrate a feature's functionality. Prefer to use `.mp4` files that are less than 512 KB. `.ogg` and `.webm` formats up to 1 MB of size are also accepted. Use of `.gif` format is deprecated and no longer permitted. |
```

Author: I'm trying to capture two separate rules in a single sentence: (1) that `.ogg` and `.webm` formats are ok, and (2) that regardless of the format, videos cannot exceed 1 MB (though less than 512 KB is preferred). 
I think the suggestion doesn't capture that they are separate rules. 

What about the following?
```suggestion
| [Animated images](media.md#animated-images) | Use short videos (of a few seconds) to demonstrate a feature's functionality. Prefer to use `.mp4` files that are less than 512 KB. `.ogg` and `.webm` formats are also accepted. The video must not exceed 1 MB of size. Use of `.gif` format is deprecated and no longer permitted. |
```

Reviewer 3: I agree with Mike's point and recommend his suggestion.
